operators/pkg/controller/remotecluster/associations.go

```diff
@@ -27,14 +29,35 @@ type associatedCluster struct {
 
 // newAssociatedCluster creates an associatedCluster from an object selector.
 func newAssociatedCluster(c k8s.Client, selector commonv1alpha1.ObjectSelector) (associatedCluster, error) {
-	ca, err := nodecerts.GetCA(c, selector.NamespacedName())
-	if err != nil {
+	var publicTransportCertsSecret v1.Secret
+	if err := c.Get(
+		transport.GetCertsPublicSecretNamespacedName(selector.NamespacedName()),
+		&publicTransportCertsSecret,
+	); err != nil {
+		if apierrors.IsNotFound(err) {
+			// TODO: would ideally return an informative error here, but our usage requires just returning a nil CA
```

Comment:
not sure what this `TODO` implies. I'd say either replace it by a log, a normal comment, change the function signature, or remove it. Otherwise this might stay there forever?

Reply:
Rewriting to just a comment.